src/planner/subquery/flatten_dependent_join.cpp

```diff
@@ -576,10 +603,73 @@ unique_ptr<LogicalOperator> FlattenDependentJoins::PushDownDependentJoinInternal
 		return plan;
 	}
 	case LogicalOperatorType::LOGICAL_RECURSIVE_CTE: {
-		throw BinderException("Recursive CTEs not (yet) supported in correlated subquery");
+		auto &setop = plan->Cast<LogicalRecursiveCTE>();
+
+#ifdef DEBUG
+		plan->children[0]->ResolveOperatorTypes();
+		plan->children[1]->ResolveOperatorTypes();
+#endif
+		plan->children[0] =
+		    PushDownDependentJoinInternal(std::move(plan->children[0]), parent_propagate_null_values, lateral_depth);
+		base_binding.table_index = setop.table_index;
+		base_binding.column_index = setop.column_count;
+		plan->children[1] =
+		    PushDownDependentJoinInternal(std::move(plan->children[1]), parent_propagate_null_values, lateral_depth);
+		RewriteCorrelatedExpressions rewriter(this->base_binding, correlated_map, lateral_depth);
+		rewriter.VisitOperator(*plan);
+
+		RewriteCorrelatedExpressions recursive_rewriter(this->base_binding, correlated_map, lateral_depth + 1, true);
+		recursive_rewriter.VisitOperator(*plan->children[0]);
+		recursive_rewriter.VisitOperator(*plan->children[1]);
+
+#ifdef DEBUG
+		plan->children[0]->ResolveOperatorTypes();
+		plan->children[1]->ResolveOperatorTypes();
+#endif
+		// we have to refer to the recursive CTE index now
+		base_binding.table_index = setop.table_index;
```

Comment:
The recursive/materialized CTE code here looks almost identical except for these three lines, perhaps they can be merged into the same switch case with these three lines only executed for `LogicalOperatorType::LOGICAL_RECURSIVE_CTE`?

Reply:
Absolutely right. I fixed that.